packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -59,15 +64,26 @@ class ConfigChainBuilder {
     while (loc !== (loc = path.dirname(loc))) {
       if (!foundConfig) {
         const configLoc = path.join(loc, BABELRC_FILENAME);
-        if (exists(configLoc)) {
-          this.addConfig(configLoc);
-          foundConfig = true;
-        }
-
+        const configJSLoc = path.join(loc, BABELRC_JS_FILENAME);
         const pkgLoc = path.join(loc, PACKAGE_FILENAME);
-        if (!foundConfig && exists(pkgLoc)) {
-          foundConfig = this.addConfig(pkgLoc, "babel", JSON);
-        }
+        const configLocs = [configLoc, configJSLoc, pkgLoc];
+        const foundConfigs = configLocs.reduce((arr, config) => {
+          if (exists(config)) {
+            const configAdded = config === pkgLoc
+              ? this.addConfig(config, "babel", JSON)
+              : this.addConfig(config);
+
+            if (configAdded && arr.length) {
+              this.errorMultipleConfigs(arr.pop(), config);
+            }
+
+            arr.push(config);
```

Comment:
Looking into https://github.com/babel/babel/pull/5543#discussion_r108017994, it looks like there is a regression here. This should be

    if (configAdded) arr.push(config);

otherwise it will always stop at the first `package.json` it encounters, even if it doesn't contain a `"babel"` key. While I actually _like_ that behavior, we should probably do it on purpose instead of on accident :P Would you be up for PRing a fix, or should I?